File: services/go-nethttp/internal/config/config_test.go
package config

import (
	"testing"
	"time"
)

func TestLoadReadsBenchmarkEnvironment(t *testing.T) {
	t.Setenv("PORT", "6000")
	t.Setenv("INFERENCE_URL", "http://inference-server:9000/")
	t.Setenv("INFERENCE_TIMEOUT_MS", "1500")
	t.Setenv("INFERENCE_HTTP_CLIENTS", "8")
	t.Setenv("CPU_PASSES", "4")
	t.Setenv("MODEL_DELAY_MS", "75")
	t.Setenv("FLUSH_INTERVAL_MS", "1000")
	t.Setenv("FLUSH_PHASE_JITTER_MS", "1000")

	cfg := Load()

	if cfg.Port != 6000 {
		t.Fatalf("port = %d, want 6000", cfg.Port)
	}
	if cfg.InferenceURL != "http://inference-server:9000" {
		t.Fatalf("inference URL = %q", cfg.InferenceURL)
	}
	if cfg.InferenceTimeout != 1500*time.Millisecond {
		t.Fatalf("timeout = %s, want 1500ms", cfg.InferenceTimeout)
	}
	if cfg.InferenceHTTPClients != 8 {
		t.Fatalf("inference clients = %d, want 8", cfg.InferenceHTTPClients)
	}
	if cfg.FlushInterval != time.Second || cfg.FlushPhaseJitter != time.Second {
		t.Fatalf("flush timing = %s/%s, want 1s/1s", cfg.FlushInterval, cfg.FlushPhaseJitter)
	}
}

File: services/java-helidon-nima/src/test/java/stt/config/ConfigTest.java
package stt.config;

import static org.junit.jupiter.api.Assertions.assertEquals;

import java.time.Duration;
import java.util.Map;
import java.util.function.Function;
import org.junit.jupiter.api.Test;

final class ConfigTest {

    private static Function<String, String> env(Map<String, String> values) {
        return values::get;
    }

    @Test
    void defaultsLoadWhenEnvIsEmpty() {
        Config config = Config.load(env(Map.of()));

        assertEquals(Config.DEFAULT_PORT, config.port());
        assertEquals("http://inference-server:9000", config.inferenceUrl());
        assertEquals(Duration.ofMillis(Config.DEFAULT_INFERENCE_TIMEOUT_MS), config.inferenceTimeout());
        assertEquals(Config.DEFAULT_INFERENCE_HTTP_CLIENTS, config.inferenceHttpClients());
        assertEquals(Config.DEFAULT_CPU_PASSES, config.cpuPasses());
        assertEquals(Config.DEFAULT_MODEL_DELAY_MS, config.modelDelayMs());
        assertEquals(Duration.ofMillis(Config.DEFAULT_FLUSH_INTERVAL_MS), config.flushInterval());
        assertEquals(Duration.ZERO, config.flushPhaseJitter());
        assertEquals(Config.DEFAULT_PARTIAL_CHANNEL_DEPTH, config.partialChannelDepth());
    }

    @Test
    void envOverridesAreParsed() {
        Map<String, String> raw = Map.of(
                "PORT", "5050",
                "INFERENCE_URL", "http://example.com:9001/",
                "INFERENCE_TIMEOUT_MS", "3000",
                "INFERENCE_HTTP_CLIENTS", "8",
                "CPU_PASSES", "8",
                "MODEL_DELAY_MS", "100",
                "FLUSH_INTERVAL_MS", "500",
                "FLUSH_PHASE_JITTER_MS", "250",
                "PARTIAL_CHANNEL_DEPTH", "16");
        Config config = Config.load(env(raw));

        assertEquals(5050, config.port());
        assertEquals("http://example.com:9001", config.inferenceUrl());
        assertEquals(Duration.ofSeconds(3), config.inferenceTimeout());
        assertEquals(8, config.inferenceHttpClients());
        assertEquals(8, config.cpuPasses());
        assertEquals(100L, config.modelDelayMs());
        assertEquals(Duration.ofMillis(500), config.flushInterval());
        assertEquals(Duration.ofMillis(250), config.flushPhaseJitter());
        assertEquals(16, config.partialChannelDepth());
    }

    @Test
    void unparseableValuesFallBackToDefaults() {
        Config config = Config.load(env(Map.of("PORT", "not-a-number", "CPU_PASSES", "")));
        assertEquals(Config.DEFAULT_PORT, config.port());
        assertEquals(Config.DEFAULT_CPU_PASSES, config.cpuPasses());
    }
}
